src/date-picker/date-picker.jsx

```diff
@@ -113,18 +116,38 @@ let DatePicker = React.createClass({
     return this.state.date;
   },
 
+  /**
+   * Setting the state will update the date here and also re-render the TextField with
+   * the new value.
+   */
   setDate(d) {
     this.setState({
       date: d,
     });
-    if (!this._isControlled()) {
-      this.refs.input.setValue(this.props.formatDate(d));
-    }
+  },
+
+  /**
+   * Open the date-picker dialog programmatically from a parent.
+   */
+  openDialog() {
+    this.setState({
+      dialogDate: this.getDate(),
+      // State changes aren't always handled immediately,
+      // better to wait on the callback.
+    }, () => this.refs.dialogWindow.show());
+  },
```

Comment:
I think this can be simplified to:
```
this.setState({
  dialogDate: this.getDate(),
  // State changes aren't always handled immediately,
  // better to wait on the callback.
}, this.refs.dialogWindow.show);
```